### ISSUE TITLE
Create the FunctionalMailboxRepository API and its memory implementation

### ISSUE THREAD

**Issue by Arsnael** [linagora/tmail-backend#254]
A functional mailbox is a mailbox that we could define like this:

```
FunctionalMailbox:
 - NAMESPACE: #functional
 - USER: domain name
 - NAME: [local part]
```

A FM would have 2 mailboxes associated to it: INBOX and Sent. Their paths would be defined like this, with the above definition, if we want to create a FM for let's say `james@linagora.com`:
```
INBOX: #functional:linagora.com:james.INBOX
Sent: #functional:linagora.com:james.Sent
```

To be able to do operations on those new FMs, we need a `FunctionalMailboxRepository` API, defined like this:
```
trait FunctionalMailboxRepository {
   def createFunctionalMailbox(f: FunctionalMailbox): Publisher[Void]
   def deleteFunctionalMailbox(f: FunctionalMailbox): Publisher[Void]
   def listFunctionalMailboxes(domain: Domain): Publisher[FunctionalMailboxes]
   def addMember(f: FunctionalMailbox, u: User): Publisher[Void]
   def removeMember(f: FunctionalMailbox, u: User): Publisher[Void]
   def listMembers(f: FunctionalMailbox): Publisher[Void]
}
```

**DoD**: Memory implementation + unit tests

**Comment by chibenwa** [comment]
Euh I thougth we said that we would directly be using MailboxManager to store functional mailboxes, without the need of other storage implementations.

Can you update this ticket accordingly?

**Comment by Arsnael** [comment]
Yeah I hesitated again when rewriting this I don't know why... Edited

**Comment by chibenwa** [comment]
Title too

**Comment by Arsnael** [comment]
I don't see how the title is wrong though?

**Comment by vttranlina** [comment]
> we would directly be using MailboxManager to store functional mailboxes, without the need of other storage implementations.

How can we do that? at least with the addMember, removeMember, listMember?

**Comment by Arsnael** [comment]
For members we would rely on ACLs, as described in the ticket?

**Comment by chibenwa** [comment]
Added `def listTeamMailboxes(user: Username): Publisher[TeamMailboxes]` method!
